### DISCUSSION TITLE
One mesh for time dependent variable

### DISCUSSION THREAD

**Discussion by Uswald** [LLNL/Silo#344] [Q&A]
I've got a time dependent vector field on a unstructured mesh. The mesh is not time dependent. I want to store the field at eac time iteration. So when I write the field I write the mesh also. Is it possible to store the mesh once and not at each iteration? If yes, how can it be done?

**Comment by markcmiller86** [comment]
Yes, I believe this is possible. It requires some care so that down-stream consumers (e.g. something like VisIt) can operate *normally* without having to do anything special to read and consumer your data.

I think this may be possible *only* on the PDB driver. So, put the mesh in one file like maybe `mesh.silo` with name `vecmesh`. Then, put the time varying vector field in each of their own files like maybe `vec000.silo`, `vec001.silo` etc. When you call `DBPutUcdvar()` to put the vector field in those files, specify the meshname as `mesh.silo:vecmesh`.

I don't think the HDF5 driver was designed to support the `<file>:<objname>` referencing mechanism.

I believe that will work. I should create an example test case for this.

**Comment by Uswald** [comment]
I did what you proposed, but when I open the results in Visit the following error happen: "There was an error opening vec*.silo database. It may be an invalid file. VisIt tried using the following file format readers to open the file: Silo"

**Reply by markcmiller86** [reply]
Can you attach file?

**Reply by Uswald** [reply]
Here is the mesh file and the vector field file.
[Coaxial_Files.zip](https://github.com/LLNL/Silo/files/12671530/Coaxial_Files.zip)
